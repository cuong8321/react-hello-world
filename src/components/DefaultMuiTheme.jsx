import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import {darkBaseTheme as defaultTheme} from 'material-ui/styles/baseThemes/darkBaseTheme'
import getMuiTheme from 'material-ui/styles/getMuiTheme'

export default function DefaultMuiTheme (props) {
  return <MuiThemeProvider muiTheme={getMuiTheme(defaultTheme)}>
    {props.children}
  </MuiThemeProvider>
}
