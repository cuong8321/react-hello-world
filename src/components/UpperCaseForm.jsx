import React from 'react'
import TextField from 'material-ui/TextField'
import FlatButton from 'material-ui/FlatButton'
import Paper from 'material-ui/Paper'

export default class UpperCaseForm extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      text: ''
    }
  }

  render () {
    const onChange = (event, text) => this.setState({text})
    const clear = () => this.setState({text: ''})

    return <Paper zDepth={1}><div>
      <div>
        <TextField id='input' hintText='Your Text Here...' onChange={onChange} value={this.state.text} />
        <FlatButton secondary label='Clear' onClick={clear} />
        <Paper zDepth={0}>
          <label htmlFor='output-upper'>Upper Case: </label>
          <output id='output-upper'>{this.state.text.toUpperCase() || <i>(Empty)</i>}</output>
        </Paper>
        <Paper zDepth={0}>
          <label htmlFor='output-lower'>Lower Case: </label>
          <output id='output-lower'>{this.state.text.toLowerCase() || <i>(Empty)</i>}</output>
        </Paper>
      </div>
      <div>
        <Paper zDepth={0} />
      </div>
    </div></Paper>
  }
}
