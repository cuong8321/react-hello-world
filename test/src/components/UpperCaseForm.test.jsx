import './__lib__/material-ui-mocks.jsx'
import React from 'react'
import renderer from 'react-test-renderer'
import UpperCaseForm from '../../../src/components/UpperCaseForm.jsx'
import DefaultMuiTheme from '../../../src/components/DefaultMuiTheme.jsx'

test('Snapshot: components/UpperCaseForm.jsx', () => {
  const tree = renderer.create(<UpperCaseForm />).toJSON()
  expect(tree).toMatchSnapshot()
})
