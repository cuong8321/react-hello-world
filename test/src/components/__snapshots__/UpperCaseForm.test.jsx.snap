// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/UpperCaseForm.jsx 1`] = `
<mocked-component
  zDepth={1}
>
  <div>
    <div>
      <mocked-component
        hintText="Your Text Here..."
        id="input"
        onChange={[Function]}
        value=""
      />
      <mocked-component
        label="Clear"
        onClick={[Function]}
        secondary={true}
      />
      <mocked-component
        zDepth={0}
      >
        <label
          htmlFor="output-upper"
        >
          Upper Case: 
        </label>
        <output
          id="output-upper"
        >
          <i>
            (Empty)
          </i>
        </output>
      </mocked-component>
      <mocked-component
        zDepth={0}
      >
        <label
          htmlFor="output-lower"
        >
          Lower Case: 
        </label>
        <output
          id="output-lower"
        >
          <i>
            (Empty)
          </i>
        </output>
      </mocked-component>
    </div>
    <div>
      <mocked-component
        zDepth={0}
      />
    </div>
  </div>
</mocked-component>
`;
